0101-nix formatting.md

```diff
@@ -0,0 +1,65 @@
+---
+feature: nix_formatting
+start-date: 2021-08-17
+author: Raphael Megzari
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+
+[summary]: #summary
+
+Decide on a recommended automated formatter for nix files in nixpkgs.
+
+# Motivation
+
+[motivation]: #motivation
+
+Prevent debate around how things should be formatted.
+
+# Detailed design
+
+[design]: #detailed-design
+
+The implementation of this RFC should include several parts
+
+- Agree on an automated formatter. [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt) is proposed here.
+- If adopted, agree on a schedule to format nixpkgs codebase. Should it be done at once, or on a per package basis.
+- Potentially agree on a hook to enforce formatting
+
+# Examples and Interactions
+
+[examples-and-interactions]: #examples-and-interactions
+
+This section is not needed for this RFC.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+- Having a commit that changes the formatting, can make git blame harder to use. It will need `--ignore-rev` flag.
+- Every formatter will have bugs. An example of a bug for nixpkgs-fmt can be found [here](https://github.com/NixOS/nixpkgs/pull/129392)
+
+# Alternatives
+
+[alternatives]: #alternatives
+
+- Alternative formatter [nixfmt](https://github.com/serokell/nixfmt)
+- Keep the status quo of not having an official formatter. The danger is that this creates discord within the nix community. On top of fragmented the community it can generate lengthy discussions (which do not advance the eco-system).
```

Comment:
There is also an alternative of explicitly discouraging any comments on formatting not directly based on coding style recommendations spelled out in the Nixpkgs manual.

Reply:
I'd say that it would be step in wrong direction. As far as I remember, nothing in nixpkgs manual says that your expression should contain newlines at all.

Reply:
Well, discouraging is not the same as strictly forbidding, if you in good faith consider something an _exceptionally_ unfortunate formatting decision, you could comment about that… with a link to a PR to the manual, obviously.

Reply:
Where would you draw the line? It should be explicit, otherwise one reviewer thinks one part is exceptionally unfortunate and the next thinks another part is exceptionally unfortunate.